xdsl/transforms/lower_csl_stencil.py

```diff
@@ -0,0 +1,219 @@
+from dataclasses import dataclass
+
+from xdsl.context import MLContext
+from xdsl.dialects import arith, func
+from xdsl.dialects.builtin import (
+    FunctionType,
+    IndexType,
+    IntegerAttr,
+    ModuleOp,
+    UnrealizedConversionCastOp,
+    i16,
+)
+from xdsl.dialects.csl import csl, csl_stencil, csl_wrapper
+from xdsl.ir import Block, Operation, Region
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+
+
+def get_dir_and_distance_ops(
+    op: csl_stencil.AccessOp,
+) -> tuple[csl.DirectionOp, arith.Constant]:
+    """
+    Given an access op, return the distance and direction, assuming as access
+    to a neighbour (not self) in a star-shape pattern
+    """
+
+    offset = tuple(op.offset)
+    assert len(offset) == 2, "Expecting 2-dimensional access"
+    assert (offset[0] == 0) ^ (
```

Comment:
I tend to find `!=` to be more readable than `^` when it comes to booleans

Reply:
Done as suggested.